src/export/digitalPlanning/model.ts

```diff
@@ -591,134 +622,51 @@ export class DigitalPlanning {
-            `details.vehicleParking.${vehicle}.count.proposed`,
-            this.passport.data?.[`proposal.${vehicle}.number.proposed`] || 0,
-          );
-          set(
-            baseProposal,
-            `details.vehicleParking.${vehicle}.offStreet.count.existing`,
-            this.passport.data?.[
-              `proposal.${vehicle}.offStreet.number.existing`
-            ] || 0,
-          );
-          set(
-            baseProposal,
-            `details.vehicleParking.${vehicle}.offStreet.count.proposed`,
-            this.passport.data?.[
-              `proposal.${vehicle}.offStreet.number.proposed`
-            ] || 0,
-          );
-          set(
-            baseProposal,
-            `details.vehicleParking.${vehicle}.onStreet.count.existing`,
-            this.passport.data?.[
-              `proposal.${vehicle}.onStreet.number.existing`
-            ] || 0,
-          );
-          set(
-            baseProposal,
-            `details.vehicleParking.${vehicle}.onStreet.count.proposed`,
-            this.passport.data?.[
-              `proposal.${vehicle}.onStreet.number.proposed`
-            ] || 0,
-          );
-        });
-
-        const carTypes = ["club", "disabled", "other", "residents"];
-        carTypes.forEach((type) => {
-          set(
-            baseProposal,
-            `details.vehicleParking.cars.offStreet.${type}.count.existing`,
-            this.passport.data?.[
-              `proposal.cars.offStreet.${type}.number.existing`
-            ] || 0,
-          );
-          set(
-            baseProposal,
-            `details.vehicleParking.cars.offStreet.${type}.count.proposed`,
-            this.passport.data?.[
-              `proposal.cars.offStreet.${type}.number.proposed`
-            ] || 0,
-          );
-          set(
-            baseProposal,
-            `details.vehicleParking.cars.onStreet.${type}.count.existing`,
-            this.passport.data?.[
-              `proposal.cars.onStreet.${type}.number.existing`
-            ] || 0,
-          );
-          set(
-            baseProposal,
-            `details.vehicleParking.cars.onStreet.${type}.count.proposed`,
-            this.passport.data?.[
-              `proposal.cars.onStreet.${type}.number.proposed`
-            ] || 0,
-          );
-        });
-      }
+    // Prior Approvals will use London Data Hub in future, but don't yet https://editor.planx.uk/opensystemslab/prior-approval-more-information
+    if (
+      this.passport.data?.["property.region"]?.[0] === "London" &&
+      !this.passport.data?.["application.type"]?.[0]?.startsWith("pa")
+    ) {
+      PARKING_TYPES.forEach((type) => {
+        set(
+          baseProposal,
+          `parking.${type}.count`,
+          this.passport.data?.[`proposal.parking.${type}`] || 0,
+        );
+        set(
+          baseProposal,
+          `parking.${type}.difference`,
+          this.passport.data?.[`proposal.parking.${type}.difference`] || 0,
+        );
+      });
```

Comment:
We have a bit on an interesting predicament here - 
- We have a new schema structure that counts certain types of parking that we want to fulfill on payload generation
- I've updated the passport mappings to reflect the _new_ content that has been written to capture this
- Even once the new content is published in the London Data Hub, there's going to be a time lag while parent services get published and existing sessions could still ask/capture the _old_ passport variables 
- At first I thought we'd perhaps be able to have multiple levels of fallbacks here (eg `"new.passport.var" || "old.passport.var" || 0`), but the structures are actually _counting different things_ and I'm not sure we can make that comparison so simply??

So, are we okay understanding that any services slow to publish new LDH changes won't have structured parking data, but those questions and answers would still come through in `responses` ? BOPS was quite clear they aren't doing anything with this yet, so hopefully no major issues?

Reply:
Yeah this is an interesting one and one that will appear a few times no doubt.

If we're counting different things, the fallback isn't a viable option I agree. I think as the data isn't currently being consumed, the approach outlined here is the right way forward 👍 

Hypothetically, if BOPS were consuming this, it gets more interesting!

I think we'd have to make it initially an optional field, wait until services updated, and then update schema to make it mandatory once that was done? Or ensure all services are updated first (hard to coordinate!) and then roll out changes to schema?

I think in an ideal world we'd want it always to be schema led. If BOPS need something, we add to schema, and then PlanX services have to "catch up" and fail validation/publishing (somehow) until they're compliant. An interesting one to mull over.

Reply:
Slightly tangential, but I was using this query the other day to chec kout Pay nodes which didn't have an update value for `govPayMetadata`.

Helpful pattern for checking if values are set, could be applied to `fn` fields in future?

```sql
SELECT 
    obj.key AS pay_key, 
    obj.value AS pay_node,
    flows.id as flow_id
FROM 
    flows,
    jsonb_each(flows.data) AS obj
WHERE 
    jsonb_typeof(obj.value) = 'object'
    AND (obj.value->>'type')::int = 400
    AND (obj.value->'data'->>'govPayMetadata') IS NULL;
```